### PR TITLE
Router, refactor

### PR DESCRIPTION
**Pull request by Yahasana**
fastRoute is not fast and smart enough

### CODE REVIEW

**Comment by mruz** [comment]
Is it that fast as a current FastRoute? http://nikic.github.io/2014/02/18/Fast-request-routing-using-regular-expressions.html

File: ice/mvc/route.zep [hunk 1]
```diff
@@ -1,51 +1,475 @@
...
+		return self::routes;
+	}
+
+	/**
+	 * Get the name of a route.
+	 *
+	 * @param   object  Route instance
+	 * @return  string
+	 */
+	public static function name(Route route)
```
Comment (mruz) [inline ice/mvc/route.zep]: `<Route> route`

**Review by mruz — COMMENTED** [review]
It would be good to have both, then we could compare the speed.

**Comment by Yahasana** [comment]
can we tag this at next release 1.4?

this will not break any compatibility but add enhancements and a new reverse route feature.
```php
// route: blog/post/10
$uri = $route->uri(["controller" => "blog", "action" => "post", "id" => 10]);

**Comment by Yahasana** [comment]
@mruz would u please check if the zephir parser is the latest version? i saw it claim support static property but the travis always show syntex error in `protected static routes = [];`

udpate: it's latest version
> $ composer install --prefer-source --no-scripts
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 29 installs, 0 updates, 0 removals
>  - Installing iceframework/zephir-parser (1.1.2): Cloning 240d8a5c9c from cache
>  - Installing phalcon/zephir (0.11.0): Cloning f5a5a14837 from cache

**Comment by Yahasana** [comment]
make a cleaner pr

**Pull request closed without merging**